website/database.py

```diff
@@ -775,12 +779,21 @@ def get_quiz_stats(self, ids, start=None, end=None):
 
     def add_program_stats(self, id, level, number_of_lines, exception):
         key = {"id#level": f"{id}#{level}", "week": self.to_year_week(date.today())}
-
         add_attributes = {"id": id, "level": level, "number_of_lines": number_of_lines}
+        p_stats = PROGRAM_STATS.get_many({"id": id, "week": self.to_year_week(date.today())})
+
+        chart_history = []
+        if p_stats.records and 'chart_history' in p_stats.records[0].keys():
+            chart_history = p_stats.records[0]['chart_history']
+        slice = MAX_CHART_HISTORY_SIZE if len(chart_history) > MAX_CHART_HISTORY_SIZE else 0
+
         if exception:
             add_attributes[exception] = dynamo.DynamoIncrement()
+            new_chart_history = chart_history + [0]
         else:
             add_attributes["successful_runs"] = dynamo.DynamoIncrement()
+            new_chart_history = chart_history + [1]
```

Comment:
same here, if chart_history would be a string it would fail. First cast to a list maybe?